src/Cart.php

```diff
@@ -831,7 +831,12 @@ private function isMulti($item)
      */
     private function storedCartWithIdentifierExists($identifier)
     {
-        return $this->getConnection()->table($this->getTableName())->where('identifier', $identifier)->exists();
+        $data = ['identifier' => $identifier];
+        if ($this->countInstances() > 1) {
```

Comment:
`countInstances` does count the Items in the cart. This methods should be called `countItems()` imo.
Also why is it relevant how many items the cart has for checking the configured instance, for example if i start with an empty cart?

Reply:
I agree with you that the `countInstances` method should be called `countItems`. The name is quite misleading. The idea was to check the number of instances being set up by the user. I could have done ` $this->getConnection()->table($this->getTableName())->where(['identifier' => $identifier, 'instance'=> $this->currentInstance()])->exists()` but this breaks some tests